UI/NpcList/NpcList.gd

```diff
@@ -6,21 +6,95 @@ onready var nameButton = $PanelContainer/VBoxC/UpperPanel/HBoxContainer/Name
 var _personalityBtnState: bool = true
 var _childrenBtnState: bool = true
+var _IDtoForget
+var nodeToFree
 
 
-func addRow(name: String, gender: String, subbyStat: float, ID: String, children: int = 0):
+func addRow(name: String, gender: String, subbyStat: float, ID: String, occupation: String, children: int = 0):
 	var newRow = npcRow.instance()
 	container.add_child(newRow)
-	newRow.initData(name, gender, subbyStat, ID, children)
+	newRow.initData(name, gender, subbyStat, ID, occupation, children)
 	newRow.connect("onForgetButtonPressed", self, "forgetNPC")
+	newRow.connect("onMeetButtonPressed", self, "meetNPC")
+	
+
+func forgetNPC(ID, name, node):
+	_IDtoForget = ID
+	nodeToFree = node
+	sendPopupMessage("Are you sure you want to forget "+name+"? This will permanently remove "+name+" from the game. Keep in mind that if this character is pregnant, their pregnancy will be forgotten too. But any kids you had together will stay.", true)
+
+
+func _on_Forget_pressed():
+	nodeToFree.queue_free()
+	nodeToFree = null
+	GM.main.removeDynamicCharacter(_IDtoForget)
+	hideNotificationButnsAndWindow()
+
+
+func _on_Cancel_pressed():
+	hideNotificationButnsAndWindow()
+
+
+func meetNPC(ID, occupation):
+	var room = GM.world.getRoomByID(GM.pc.getLocation())
+	match occupation:
+		"Inmates": 
+			if(WorldPopulation.Inmates in GM.pc.getLocationPopulation()):
+				GM.main.runScene("InmateExposureForcedSexScene", [ID])
+				GM.main.runCurrentScene()
+			else:
+				sendPopupMessage("There are no inmates in this location.\nTry looking elswhere")
+		"Guards":
+			if(WorldPopulation.Guards in GM.pc.getLocationPopulation()):
+			#if(room.loctag_GuardsEncounter || room.loctag_Greenhouses):
+				GM.main.runScene("GuardCaughtOfflimitsScene", [ID])
+				GM.main.runCurrentScene()
+			else:
+				sendPopupMessage("There are no guards in this location.\nTry looking at the checkpoint or near greenhouses")
+		"Engineers":
+			if(room.loctag_EngineersEncounter):
+				GM.main.runScene("EngineerCaughtOfflimitsScene", [ID])
+				GM.main.runCurrentScene()
+			else:
+				sendPopupMessage("There are no engineers in this location.\nTry looking at the engineering bay")
+		"Nurses":
+			if(room.loctag_MentalWard):
+				GM.main.runScene("NurseCaughtOfflimitsScene", [ID])
+				GM.main.runCurrentScene()
+			else:
+				sendPopupMessage("There are no nurses in this location.\nTry looking at the restricted area of the mediacl ward")
+
+
+func sendPopupMessage(msgText: String = "", isForgetWindow: bool = false):
+	if(isForgetWindow):
+		popupForgetBtn.visible = true
+		popupCancelBtn.visible = true
+		popupWindowLabel.text = msgText
+		popupWindow.popup_centered_ratio(0.3)
+	else:
+		popupOkBtn.visible = true
+		popupWindowLabel.text = msgText
+		popupWindow.popup_centered_ratio(0.3)
+
+
+func _on_Ok_pressed():
+	hideNotificationButnsAndWindow()
+
 
-# add pop-up confirmation window
-func forgetNPC(ID):
-	GM.main.removeDynamicCharacter(ID)
+func hideNotificationButnsAndWindow():
```

Comment:
technically you don't need this function `popupWindow.visible = false` would be enough ^^. If you still want this function I'd suggest naming it `hideNotificationWindow`. Also don't shorten the words like `Butns`, looks better if you don't ^^